### ISSUE TITLE
[java] InvalidLogMessageFormat may examine the value of a different but identically named String variable

### ISSUE THREAD

**Issue by krallus** [pmd/pmd#3284]
**Affects PMD Version: 6.33.0 (via Eclipse PMD For Eclipse 4.22.0)** 

**Rule: [InvalidLogMessageFormat](https://pmd.github.io/pmd-6.33.0/pmd_rules_java_errorprone.html#invalidlogmessageformat)**

**Description:**

InvalidLogMessageFormat does not respect the scope in which a String format variable is declared and may use the value of an identically named variable that has been declared in a different scope.

**Code Sample demonstrating the issue:**

```
	public void aTest() {
		if (true) {
			final String logMessage = "A message formatted with three parameters: {}, {}, {}";
		}

		if (true) {
			final String logMessage = "A message formatted with only one parameter: {}";
			final Object param = null;
			logger.trace(logMessage, param);
		}
	}
```

**Expected outcome:**

PMD reports a violation at line 8, but that's wrong. That's a false positive.

Specifically, the error is reported as:
_InvalidLogMessageFormat: Missing arguments, expected 3 arguments but have 1_

**Running PMD through:** *Eclipse*
